sklearn/preprocessing/_encoders.py

```diff
@@ -243,6 +261,9 @@ class OneHotEncoder(_BaseEncoder):
         - array : ``drop[i]`` is the category in feature ``X[:, i]`` that
           should be dropped.
 
+        If there are infrequent categories and `drop` selects any of the
+        infrequent categories, then all these categories are dropped.
```

Comment:
> if someone specifies "sklearn_infrequent" as the one to drop?

Focusing on the use case where `drop` is a array-like of categories to drop, then `infrequent_sklearn` becomes a place holder for "drop infrequent". The string will have different semantics compared to all other strings in `drop`. 

Lets say `drop=["a", "dog", "infrequent_sklearn"]`, the first two features will drop the actual categories "a" and "dog", while the last feature will drop the infrequent categories.

Maybe we can add another parameter, "infrequent_category_name" to specific what the name of the infrequent category is. I really do not want to add another parameter, but it will help with edge cases with `drop` or if `infrequent_sklearn` is an actual category.

1. For `drop`, if `infrequent_category_name='infrequent_sklearn'` (default) and drop has "infrequent_sklearn", then we know to drop the infrequent category
2. If `infrequent_sklearn` is an actual category, then `infrequent_category_name` can be set to another name. This also helps with `feature_names_out`.